### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by masonnl**
This branch is for general purpose integration tests using Poltergeist as the headless browser via PhantomJS. @HParker and @CRiva note that in order for the suite to run, all machines must have [phantomJS](http://phantomjs.org/) installed. It's as simple as `apt-get install phantomjs`.

This PR includes:
- An `integration/` directory that will house the poltergeist driver
- A `page_objects/` directory that will house PageObject(s)
    - Page Objects will represent objects on the page that we can test 
- A `spec/` directory where we can house tests for the Horizon

More to come...

### CODE REVIEW

File: spec/integration/page_objects/page_object.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+require_relative '../poltergeist_helper'
+
+class PageObject
```
Comment (XanderStrike) [inline spec/integration/page_objects/page_object.rb]: Fuck yeah page objects.

File: test/integration/page_objects/page_object.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  #########
+  # Waits #
+  #########
+
+  def wait_for_ajax(time_out = TIMEOUT)
+    time_out.times do
+      page.evaluate_script('jQuery.active').zero? ? true : sleep(0.1)
+    end
+    raise "Ajax never finished running."
+  end
```
Comment (HParker) [inline test/integration/page_objects/page_object.rb]: Nice! I had to redescover wait for ajax at work recently. So nice to have it out the gate!

File: test/integration/spec/general/login_test.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require_relative '../../poltergeist_helper'
+
+class SignInTest < PoltergeistIntegrationTest
+
+  describe 'user' do
+    it 'can login' do
+      puts "Hello, world!"
```
Comment (HParker) [inline test/integration/spec/general/login_test.rb]: :)

**Comment by HParker** [comment]
Tests pass. I see no reason not to merge this.

**Pull request merged**